ethpm_cli/config.py

```diff
@@ -29,3 +38,43 @@ def __init__(self, args: Namespace) -> None:
             if not self.ethpm_dir.is_dir():
                 self.ethpm_dir.mkdir()
         validate_ethpm_dir(self.ethpm_dir)
+
+        # Setup xdg ethpm dir
+        xdg_ethpm_dir = get_xdg_ethpmcli_root()
+        w3 = Web3(load_provider_from_uri(INFURA_HTTP_URI))
+        if xdg_ethpm_dir.is_dir():
+            validate_xdg_ethpm_dir(xdg_ethpm_dir, w3)
+        else:
+            initialize_xdg_ethpm_dir(xdg_ethpm_dir, w3)
```

Comment:
I don't know if it's possible but if you can make `initialize_xdg_ethpm_dir` idempotent then you can simplify this to just be something that is unconditionally called at this stage and include the validation in the initialization logic.